solvation_analysis/rdf_parser.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+from scipy.interpolate import interp1d, UnivariateSpline
+import scipy
+import matplotlib.pyplot as plt
+import warnings
+
+
+def interpolate_rdf(bins, rdf, floor=0.05, cutoff=5):
+    start = np.argmax(rdf > floor)  # will return first index > rdf
+    end = np.argmax(bins > cutoff)  # will return first index > cutoff
+    bounds = (bins[start], bins[end - 1])
+    f = UnivariateSpline(bins[start:end], rdf[start:end], k=4, s=0)
+    return f, bounds
+
+
+def identify_minima(f):
+    assert isinstance(
+        f, scipy.interpolate.fitpack2.InterpolatedUnivariateSpline
+    ), "this function is compatible with the scipy.interpolate package"
+    cr_pts = f.derivative().roots()
+    cr_vals = f(cr_pts)
+    return cr_pts, cr_vals
+
+
+def plot_interpolation_fit(bins, rdf, **kwargs):
+    f, bounds = interpolate_rdf(bins, rdf, **kwargs)
+    x = np.linspace(bounds[0], bounds[1], num=100)
+    y = f(x)
+    pts, vals = identify_minima(f)
+    plt.plot(bins, rdf, "b--", label="rdf")
+    plt.plot(x, y, "r-", label="interpolation")
+    plt.plot(pts, vals, "go", label="critical points")
+    plt.xlabel("Radial Distance (A)")
+    plt.ylabel("Probability Density")
+    plt.title("Interpolation of RDF with quartic spline")
+    plt.legend()
+    plt.show()
+
+
+def identify_solvation_cutoff(
```

Comment:
AFAIK this only identifies a single solvation cutoff, this should defs be extensible to as many as you want, ie an array of minima.  This just means you need to something along the lines of:

```python 
#check that there is a min and max for each point
#this may not be true @orion and you will know if you need to check differently
assert(len(cr_points)%2 == 0)
minima = np.empty(len(cr_points)/2)
for i in range(0,len(cr_points),2):
	#assume max always comes first and that it goes max-min-max-min
	max = cr_points[i]
	min = cr_points[i+1]
	if max <= min:
		#some error path here
		raise ShockinglyHorribleInterpolationException
	else:
		minima[i] = min
#some other stuff
return minima
```
That way you can get a set of solvation critical points rather than just the first one. 
This is by no means the best solution, but I thought i would try and convey the idea in some (pseudo) code.

Reply:
I agree in general but I think I need to improve the minima identifying method before generalizing to further solvation cutoffs beyond the first. The min-max min-max approach isn't good enough to weed out very slight split peaks and small divets in the fit. I could try applying a stronger smoothing factor to the interpolation or throwing in an additional peak fitting step, but as you suggested, I think it's best to get something that works and start developing the analysis framework. 

For now, I think I should throw together a heuristic for the first solvation cutoff and move on. I can circle back to improving the fitting algorithm later. I renamed issue #19 to reflect this, I think it is closely related.

Reply:
Good decision I think to move forward while documenting the limitations. As you say I think we want to develop the analysis framework rather than focusing on the perfect minima finding heuristic.